README.md

```diff
@@ -12,28 +12,185 @@ We believe that, like native platforms, the web deserves a first-class virtual s
+* Allow contents of the scroller to work with find-in-page, accessibility features, focus, fragment URL navigation, etc., just as they would work in a non-virtualized context.
+* Be flexible enough to allow developers to integrate advanced features and behaviors, such as groupings, sticky headers, animations, swiping, item selection, etc.
+* Support 1D horizontal/1D vertical/2D wrapped-grid layouts.
 
-The [`virtual-content`](https://github.com/valdrinkoshi/virtual-scroller/tree/virtual-content) branch starts from scratch, building on the [searchable invisible DOM](https://github.com/rakina/searchable-invisible-dom) proposal. For now, it only contains the `<virtual-content>` element, which manages the searchable-invisible-ness of its children. It does not contain any notion of item, or DOM creation/updating/recycling; it operates directly on its DOM element children. Importantly, unlike traditional virtualization, all of these children are present, even if the majority of them are (searchable-)invisible.
+## Non-goals
 
-This new approach, while simple to use, is fairly powerful! Furthermore, preliminary tests and demos show that it does solve the fundamental problems mentioned above: with Blink's flagged implementation of searchable invisible DOM, the resulting virtualized content can still be found by find-in-page, or intra-page anchor navigation, or tabbed to, or found by search engine indexers. (Accessibile landmark navigation support is up next!)
+* Allow data that are not part of the DOM to work with find-in-page, accessibility, etc. The DOM remains the source of truth for these browser features, not e.g. the server-side, or other in-memory JavaScript data structures. See [below](#almost-infinite-data-from-the-server) for more elaboration.
 
-But, there is much more work to do to show that this approach is feasible. We need to re-build the demos corpus to attain parity with the `traditional-virtualization` branch, to show that all the desired user experiences and framework integrations are still possible. And we need to work out an easy-to-use solution for populating the `<virtual-content>` element's initial set of children, without causing jank.
+And, at least in v1, the following are out of scope:
 
-## Talk at Chrome Dev Summit 2018
+* Built-in support for advanced features and behaviors, such as those mentioned above.
+* Support infinite-grid (spreadsheet-like) or masonry layouts.
 
-At Chrome Dev Summit 2018, Gray Norton presented on our work so far. Give it a watch!
+## Proposed APIs
 
-[![virtual-scroller at Chrome Dev Summit 2018](http://img.youtube.com/vi/UtD41bn6kJ0/0.jpg)](http://www.youtube.com/watch?v=UtD41bn6kJ0)
+### `<virtual-content>` element
+
+The `<virtual-content>` element represents a container that will manage the rendering of its children.
+The children of this element might not get rendered/updated if they are not near or in the viewport.
```

Comment:
As a consequence of display-locking, the layout-related values might be observably different for virtualized vs non-virtualized children. Should we mention that?

Reply:
You mean, e.g., all children-of-children would have 0 width/height according to getBoundingClientRect? I will mention that in the new section I am adding. But I might have misunderstood, so please correct my text if so.

Reply:
Not exactly, it will have approximately correct width and height and offsets etc, which might be different than the actual values when they are made visible, because of the placeholder sizes. Changed the wording to "Changing the layout of non-visible children's descendants", is that to broad?

Reply:
That seems perfect!